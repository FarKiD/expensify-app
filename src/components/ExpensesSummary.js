import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
	const formattedExpensesTotal = numeral(expensesTotal).format(expensesTotal > 999999 ? '0,0a' : '0,0');
	
	return (
		<div>
			<h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal} Toman</h1>
		</div>
	);
};

const mapStateToProps = (state) => {
	const visibleExpenses = selectExpenses(state.expenses, state.filters);
	
	return {
		expenseCount: visibleExpenses.length,
		expensesTotal: expensesTotal(visibleExpenses)
	};
};

export default connect(mapStateToProps)(ExpensesSummary);