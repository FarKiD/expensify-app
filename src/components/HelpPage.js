import React from 'react';

const HelpPage = () => (
	<p>help page and other stories</p>
);

export default HelpPage;