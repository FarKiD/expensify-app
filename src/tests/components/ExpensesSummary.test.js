import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={2500} />);
	expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expense', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={20} expensesTotal={2500000000} />);
	expect(wrapper).toMatchSnapshot();
});