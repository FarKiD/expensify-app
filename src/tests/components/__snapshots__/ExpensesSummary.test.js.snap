// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    2,500
     Toman
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    20
     
    expenses
     totalling 
    3b
     Toman
  </h1>
</div>
`;
